tdrs-backend/tdpservice/users/test/test_auth.py

```diff
@@ -326,6 +328,94 @@ def test_login_with_existing_user(mocker, api_client, user):
     assert response.status_code == status.HTTP_302_FOUND
 
 
+@pytest.mark.django_db
+def test_login_with_old_email(mocker, api_client, user):
+    """Login should work with existing user."""
+    os.environ["JWT_KEY"] = test_private_key
+    user.username = "test_old_email@example.com"
+    user.save()
+    nonce = "testnonce"
+    state = "teststate"
+    code = secrets.token_hex(32)
+    mock_post = mocker.patch("tdpservice.users.api.login.requests.post")
+    token = {
+        "access_token": "hhJES3wcgjI55jzjBvZpNQ",
+        "token_type": "Bearer",
+        "expires_in": 3600,
+        "id_token": os.environ["MOCK_TOKEN"],
+    }
+    mock_decode = mocker.patch("tdpservice.users.api.login.jwt.decode")
+    decoded_token = {
+        "email": "test_new_email@example.com",
+        "email_verified": True,
+        "nonce": nonce,
+        "iss": "https://idp.int.identitysandbox.gov",
+        "sub": user.sub,
+        "verified_at": 1577854800,
+    }
+    mock_post.return_value = MockRequest(data=token)
+    mock_decode.return_value = decoded_token
+    factory = APIRequestFactory()
+    view = TokenAuthorizationOIDC.as_view()
+    request = factory.get("/v1/login", {"state": state, "code": code})
+    request.session = api_client.session
+    request.session["state_nonce_tracker"] = {
+        "nonce": nonce,
+        "state": state,
+        "added_on": time.time(),
+    }
+    response = view(request)
+    # Ensure the user's username was updated with new email.
+    assert User.objects.filter(username="test_new_email@example.com").exists()
+    assert response.status_code == status.HTTP_302_FOUND
+
+
+@pytest.mark.django_db
+def test_login_with_initial_superuser(mocker, api_client, user):
```

Comment:
This tests that the initial superuser gets the proper `sub` value after their initial login attempt.